docs/cert-manager.md

```diff
@@ -1,76 +1,448 @@
+
+### Create TLS certificate for Console service
+
+Request Cert Manager to issue a new certificate including the following DNS domains for console:
+
+```shell
+console
+console.minio-operator.svc.
+console.minio-operator.svc.<cluster domain>
+```
+
+> [!IMPORTANT]
+> Replace `<cluster domain>` with the actual values for your MinIO tenant.
+> `cluster domain` is the internal root DNS domain assigned in your Kubernetes cluster. Typically this is `cluster.local`, check on your coredns
+> configuration for the correct value for your Kubernetes cluster. For example, using `kubectl get configmap coredns -n kube-system -oyaml | yq ".data"`.
+> The way the root DNS domain is managed can vary depending on the Kubernetes distribution (Openshift, Rancher, EKS, etc.)
+
+Create a `Certificate` for the domains mentioned above:
+
+```yaml
+# console-tls-certificate.yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: console-certmanager-cert
+  namespace: minio-operator
+spec:
+  dnsNames:
+    - console
+    - console.minio-operator.svc
+    - console.minio-operator.svc.cluster.local
+  secretName: console-tls
+  issuerRef:
+    name: minio-operator-ca-issuer
+```
+
+```shell
+kubectl apply -f console-tls-certificate.yaml
+```
+
+This creates a secret called `console-tls` in the `minio-operator` namespace.
+
+> [!IMPORTANT]
+> The secret name is not optional. Make sure the secret name is `console-tls` by setting `spec.secretName: console-tls` as in the example above.
+
+### Install Operator with Auto TLS disabled for STS
+
+When installing the Operator deployment, make sure to set `OPERATOR_STS_AUTO_TLS_ENABLED: off` env variable in the `minio-operator` container. This prevents
+MinIO Operator from issuing the certificate for STS and instead wait for you to provide the TLS certificate issued by Cert Manager.
+
+Also make sure to set `MINIO_CONSOLE_TLS_ENABLE: off` env variable, this prevents MinIO Operator from issuing the certificate for Console
+and instead will expect you to provide the TLS certificate issued by Cert Manager.
+
+> [!WARNING]
+> Failing to provide the secret `sts-tls` containing the TLS certificate or providing an invalid key-pair in the secret will
+> prevent the STS service from starting.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: minio-operator
+  namespace: minio-operator
+spec:
+...
+  template:
+    metadata:
+      labels:
+        name: minio-operator
+...
+    spec:
+      serviceAccountName: minio-operator
+      containers:
+        - name: minio-operator
+          image: minio/operator:v5.0.14
+...
+          env:
+            - name: MINIO_CONSOLE_TLS_ENABLE:
+              value: "off"
+            - name: OPERATOR_STS_AUTO_TLS_ENABLED
+              value: "off"
+            - name: OPERATOR_STS_ENABLED
+              value: "on"
+```
+
+# MinIO tenant TLS certificates with cert-manager
+
+## Create tenant-1 namespace CA Issuer
```

Comment:
Perhaps suggest to deploy the tenant first, and to set:
```
kubectl patch tenant -n tenant-1 myminio --type='merge' -p '{"spec":{"requestAutoCert": false, "env": [{"name": "MINIO_BROWSER_LOGIN_ANIMATION","value": "false"}, {"name": "MINIO_CI_CD", "value": "on"}] }}'
```

Reply:
If the Tenant is deployed with defaut values `requestAutoCert` will be true,  and Operator is going to create the autocerts, this can lead to conflicts, instead the Tenant should be created with `requestAutoCert: false` from the begginging, not patching.

Reply:
Should we try to catch that with validation? that is, if cert-manager and auto-cert should not be mixed, we should reject that in some way (at least in UI if not via crd) ?

Reply:
Certificates provided in the `spec. externalCertSecret` and `spec.requestAutoCert` are not opposites, we could have `.spec.requestAutoCert: true` and yet have cert manager provision additional certificates for the Load Balancer  domain, or additional domains that MinIO needs to handle in `spec.externalCertSecret`

Is just that in this guide, where we only use cert manager `spec. requestAutoCert` is not needed, we just need to turn it off.